### DISCUSSION TITLE
what if I emit same event in on?

### DISCUSSION THREAD

**Discussion by navneetgarggate6** [miguelgrinberg/python-socketio#1051] [Q&A]
what if I wrote following code

```
@sio.on('helloTest')
def helloTest(sid, data):
    print('test0')
    sio.emit("helloTest", data)
    print('test1')
```
Is this ok ?

**Comment by miguelgrinberg** [comment]
Sure, this is fine. Events that you emit are sent to the other side of the correction. If this is the server, then this event will be sent to the client, so this isn't a recursive call, if that is what you thought.
